practical/ProcessMining/group2/conformance_checking/src/comparison.py

```diff
@@ -0,0 +1,234 @@
+
+        # Compare with original log
+        conformance_checking = ConformanceChecking()
+        result = conformance_checking.get_conformance_value(
+            footprint_of_log, footprint_of_replayed_log
+        )
+
+        return result
+
+    def model_2_model(
+        self, log: str, algorithms: List['AlgoPm4Py'], scenario: int
+    ) -> Dict[str, float]:
+        """
+        This function takes a log, a list of algorithms and one of two scenarios.
+        The algorithms are being run on the log in order to get multiple models, which are then being replayed in order to get new logs.
+        The resulting logs are compared in dfferent ways, depending on the scenario.
+        Scenario 1: Each log is compared to the original log.
+        Scenario 2: Each log is compared to the other logs.
+
+        Parameters:
+        ----------
+
+        log: str
+            Path to the event log.
+        algorithms: [AlgoPm4Py]
+            List of algorithms to be used.
+        scenario: number
+            Scenario to be used.
+
+        Returns:
+        -------
+        Dictionary containing the comparison values.
+        """
+
+        log_from_file = get_log_from_file(log)
+
+        generated_fpms = {}
+        fpm_original = None
+        for algorithm in algorithms:
+            pipeline_fpm_original, pipeline_fpm_replayed = self.pipeline(
+                log_from_file, algorithm
+            )
+            visualize_sorted_dict(
+                pipeline_fpm_replayed.relations, "m2m_{}".format(str(algorithm))
+            )
+
+            generated_fpms[str(algorithm)] = pipeline_fpm_replayed
+            if fpm_original == None:
+                fpm_original = pipeline_fpm_original
+                visualize_sorted_dict(pipeline_fpm_original.relations, "m2m_original")
+
+        comparison_values = {}
+        if scenario == 1:
+            for algorithm, fpm_from_algorithm in generated_fpms.items():
+                key = "InputLog vs. {}".format(str(algorithm))
+                print("Checking Conformance Value of: {}".format(key))
+                conformance_checking = ConformanceChecking()
+                comparison_values[key] = conformance_checking.get_conformance_value(
+                    fpm_original, fpm_from_algorithm
+                )
+
+        if scenario == 2:
+            for (algo1, fpm1), (algo2, fpm2) in combinations(generated_fpms.items(), 2):
+                key = "{} vs. {}".format(algo1, algo2)
+                print("Checking Conformance Value of: {}".format(key))
+                conformance_checking = ConformanceChecking()
+                comparison_values[key] = conformance_checking.get_conformance_value(
+                    fpm1, fpm2
+                )
+
+        return comparison_values
+
+
+log_path = "InputLogs/L8.csv"
+
+
+com = Comparison()
+print(
+    com.model_2_model(
+        log_path,
+        [AlgoPm4Py.ALPHA, AlgoPm4Py.ALPHAPLUS, AlgoPm4Py.INDUCTIVEMINER],
+        2,
+    )
+)
```

Comment:
I would put experimental runs of the logic into a separate file or Jupyter notebook, so that the class files stay clean.